### PR TITLE
[MRV2][Performance] Skip no-op FP32 logits materialization

### PR DESCRIPTION
**Pull request by jesse996**
## Purpose

MRV2 currently materializes the logits tensor as FP32 before applying sampling parameters, even when every active request is a no-op: temperature is 0 or 1, top-k equals the vocabulary size, top-p is 1, min-p is 0, and there are no penalties, logit biases, or bad words.

The Gumbel sampling kernel already converts loaded logits to FP32 in registers. This PR adds a no-op fast path that returns the original logits tensor and preserves the existing FP32 path whenever any active request needs logits processing.

For batch size 128 and vocabulary size 151,936, the fast path avoids materializing a 74.2 MiB FP32 tensor per decoding step, along with the associated copy kernel and memory traffic.

## Test Result

- `.venv/bin/python -m pytest tests/v1/worker/test_gpu_gumbel_sample.py -v`: **passed**.
---

<details>
<summary>Essential Elements of an Effective PR Description Checklist</summary>

- [x] The purpose of the PR is explained.
- [x] The test plan includes concrete commands and coverage.
- [x] The test results and environment limitation are recorded.
- [x] Documentation is not required because this is an internal performance optimization with no user-facing API change.

</details>

### CODE REVIEW

File: benchmarks/overheads/benchmark_mrv2_sampler_noop.py
Comment (yewentao256) [inline benchmarks/overheads/benchmark_mrv2_sampler_noop.py:1]: We don't need this file

File: tests/v1/worker/test_gpu_model_runner_v2_sampler.py
Comment (yewentao256) [inline tests/v1/worker/test_gpu_model_runner_v2_sampler.py:1]: We don't need this test

**Review by yewentao256 — COMMENTED** [review]
The idea looks good

Please also attach with e2e acc metrics using `lm_eval` and e2e perf result using `vllm bench serve ...`

**Comment by jesse996** [comment]
@yewentao256 hello,this is my vllm bench result
## srart server
```bash
VLLM_USE_V2_MODEL_RUNNER=1 vllm serve "$MODEL"   --host 0.0.0.0   --port 8000   --max-model-len 512   --max-num-seqs 256   --gpu-memory-utilization 0.90
```

## vllm bench
```bash
vllm bench serve   --backend openai   --base-url http://127.0.0.1:8000   --model "$MODEL" 
  --endpoint /v1/completions   --dataset-name random   --input-len 32   --output-len 256   --num-prompts 1024   --num-warmups 128   --max-concurrency 256   --request-rate inf   --ignore-eos   --temperature 1.0   --top-p 1.0   --top-k 0   --min-p 0.0   --disable-shuffle   --seed 0   --percentile-metrics ttft,tpot,itl,e2el   --metric-percentiles 50,90,95,99   --save-result   --result-dir "./after"   --result-filename run1-01.json
```

## before
IMG_2771


## after
IMG_2772

File: tests/v1/worker/test_gpu_gumbel_sample.py [hunk 1]
```diff
@@ -183,6 +183,41 @@ def test_greedy_temperature_zero_returns_argmax():
...
+        idx_mapping,
+        temp,
+        seed,
+        pos,
+        apply_temperature=False,
+    )
+
+    assert torch.equal(sampled, sampled_from_fp32)
+
+
```
Comment (yewentao256) [inline tests/v1/worker/test_gpu_gumbel_sample.py]:
```suggestion
```

Comment (yewentao256) [reply tests/v1/worker/test_gpu_gumbel_sample.py]: I don't think we need a specific unit test for this small update

**Comment by yewentao256** [comment]
Sorry but why closing this PR?

**Comment by jesse996** [comment]
> Sorry but why closing this PR?

Sorry, I encountered merge conflicts yesterday, so I closed the PR while resolving them and rerunning the tests. I have now updated the branch with the latest main.

**Comment by jesse996** [comment]
@yewentao256 pls help to review this

**Review by yewentao256 — APPROVED** [review]
LGTM, thanks for the work!

**Comment by jesse996** [comment]
@yewentao256 hello,these failed testings don’t appear to be related to this PR, could you take a look and merge?

**Review by yewentao256 — COMMENTED** [review]
Let's retry once before a force merge

**Comment by jesse996** [comment]
@yewentao256 hello, I’ve seen the same test failure in other PRs as well, such as #49857 , so it doesn’t appear to be related to my changes.

**Review by yewentao256 — COMMENTED** [review]
I retried the CI, if still fails, please ask a force merge in slack "PR merge requests" channel

**Comment by jesse996** [comment]
> I retried the CI, if still fails, please ask a force merge in slack "PR merge requests" channel

thank you！i will do this next time～
